ksmt-cvc5/src/main/kotlin/org.ksmt.solver.cvc5/KCvc5ExprInternalizer.kt

```diff
@@ -0,0 +1,1028 @@
+        transformArray(args) { args -> nsolver.mkTerm(Kind.ADD, args) }
+    }
+
+    override fun <T : KArithSort> transform(expr: KSubArithExpr<T>) = with(expr) {
+        transformArray(args) { args -> nsolver.mkTerm(Kind.SUB, args) }
+    }
+
+    override fun <T : KArithSort> transform(expr: KMulArithExpr<T>) = with(expr) {
+        transformArray(args) { args -> nsolver.mkTerm(Kind.MULT, args) }
+    }
+
+    override fun <T : KArithSort> transform(expr: KUnaryMinusArithExpr<T>) = with(expr) {
+        transform(arg) { arg: Term -> nsolver.mkTerm(Kind.NEG, arg) }
+    }
+
+    override fun <T : KArithSort> transform(expr: KDivArithExpr<T>) = with(expr) {
+        transform(lhs, rhs) { lhs: Term, rhs: Term -> nsolver.mkTerm(Kind.DIVISION, lhs, rhs) }
+    }
+
+    override fun <T : KArithSort> transform(expr: KPowerArithExpr<T>) = with(expr) {
+        transform(lhs, rhs) { lhs: Term, rhs: Term -> nsolver.mkTerm(Kind.POW, lhs, rhs) }
+    }
+
+    override fun <T : KArithSort> transform(expr: KLtArithExpr<T>) = with(expr) {
+        transform(lhs, rhs) { lhs: Term, rhs: Term ->
+            nsolver.mkTerm(Kind.LT, lhs, rhs)
+        }
+    }
+
+    override fun <T : KArithSort> transform(expr: KLeArithExpr<T>) = with(expr) {
+        transform(lhs, rhs) { lhs: Term, rhs: Term ->
+            nsolver.mkTerm(Kind.LEQ, lhs, rhs)
+        }
+    }
+
+    override fun <T : KArithSort> transform(expr: KGtArithExpr<T>) = with(expr) {
+        transform(lhs, rhs) { lhs: Term, rhs: Term ->
+            nsolver.mkTerm(Kind.GT, lhs, rhs)
+        }
+    }
+
+    override fun <T : KArithSort> transform(expr: KGeArithExpr<T>) = with(expr) {
+        transform(lhs, rhs) { lhs: Term, rhs: Term ->
+            nsolver.mkTerm(Kind.GEQ, lhs, rhs)
+        }
+    }
+
+    override fun transform(expr: KModIntExpr) = with(expr) {
+        transform(lhs, rhs) { lhs: Term, rhs: Term ->
+            nsolver.mkTerm(Kind.INTS_MODULUS, lhs, rhs)
+        }
+    }
+
+    // custom implementation
+    override fun transform(expr: KRemIntExpr) = with(expr) {
+        transform(lhs, rhs) { lhs: Term, rhs: Term ->
+            // there is no ints remainder in cvc5
+            val remSign = nsolver.mkTerm(Kind.GEQ, lhs, zeroIntValueTerm)
+                .xorTerm(nsolver.mkTerm(Kind.GEQ, rhs, zeroIntValueTerm))
+            val modTerm = nsolver.mkTerm(Kind.INTS_MODULUS, lhs, rhs)
+
+            remSign.iteTerm(nsolver.mkTerm(Kind.NEG, modTerm), modTerm)
+        }
+    }
+
+    override fun transform(expr: KToRealIntExpr) = with(expr) {
+        transform(arg) { arg: Term ->
+            nsolver.mkTerm(Kind.TO_REAL, arg)
+        }
+    }
+
+    override fun transform(expr: KInt32NumExpr) = with(expr) {
+        transform { nsolver.mkInteger(expr.value.toLong()) }
+    }
+
+    override fun transform(expr: KInt64NumExpr) = with(expr) {
+        // We need to pass String value here because on Windows it might be cut to 32 bit int value
```

Comment:
Do we have this problem in `mkBitVector`?

Reply:
No, we don't